File: nodejs/cucumber_html/model/catalog.ts
import lodash from 'lodash';
import axios from 'axios';


const url = "https://api.restful-api.dev/objects";

const headers = {
  'Content-Type': 'application/json',
  'accept': 'application/json'
};

class CatalogObject {

  id: string;
  name: string;
  data: object;

  constructor(obj: object) {
    this.id = obj['id'];
    this.name = obj['name'];
    this.data = lodash.cloneDeep(obj['data']);
  }
}

class Catalog {

  async get(id: string, world: any) {
    await world.attach(`GET ${url}/${id}`, "text/plain");
    return await axios.get(`${url}/${id}`);
  }

  async post(payload: object, world: any) {
    world.attach(`POST ${url}`, "text/plain");
    world.attach(JSON.stringify(payload), "application/json");
    return await axios.post(url, payload);
  }

  async delete(id: string, world: any) {
    await world.attach(`DELETE ${url}/${id}`, "text/plain");
    return await axios.delete(`${url}/${id}`);
  }

}

export { Catalog, CatalogObject };

File: python/pytest/rest_api/model/catalog.py
import json
import requests

base_url = "https://api.restful-api.dev/objects"

headers = {
    'Content-Type': 'application/json',
    'accept': 'application/json'
}

def get(id):
    url = f"{base_url}/{id}"
    result = requests.get(url=url, headers=headers)
    return result


def post(payload):
    url = f"{base_url}"
    result = requests.post(
        url=url,
        data=json.dumps(payload),
        headers=headers
    )
    return result


def delete(id):
    url = f"{base_url}/{id}"
    result = requests.delete(url=url)
    return result
